tests/chainer_tests/functions_tests/pooling_tests/test_unpooling_nd.py

```diff
@@ -0,0 +1,257 @@
+            y_expected, y.data, **self.check_forward_options)
+
+    @condition.retry(3)
+    def test_forward_cpu(self):
+        self.check_forward(self.x)
+
+    @attr.gpu
+    @condition.retry(3)
+    def test_forward_gpu(self):
+        self.check_forward(cuda.to_gpu(self.x))
+
+    def check_forward_consistency_regression(self, x_data):
+        # Regression test to two-dimensional unpooling layer.
+
+        if len(self.dims) != 2:
+            return
+
+        ksize = self.ksize
+        stride = self.stride
+        pad = self.pad
+
+        y_nd = functions.unpooling_nd(x_data, ksize, stride=stride, pad=pad,
+                                      cover_all=self.cover_all)
+        y_2d = functions.unpooling_2d(x_data, ksize, stride=stride, pad=pad,
+                                      cover_all=self.cover_all)
+        testing.assert_allclose(
+            y_nd.data, y_2d.data, **self.check_forward_options)
+
+    @condition.retry(3)
+    def test_forward_consistency_regression_cpu(self):
+        self.check_forward_consistency_regression(self.x)
+
+    @attr.gpu
+    @condition.retry(3)
+    def test_forward_consistency_regression_gpu(self):
+        self.check_forward_consistency_regression(cuda.to_gpu(self.x))
+
+    def check_backward(self, x_data, y_grad):
+        ndim = len(self.dims)
+        gradient_check.check_backward(
+            functions.UnpoolingND(ndim, self.ksize, self.stride, self.pad,
+                                  cover_all=self.cover_all),
+            x_data, y_grad, **self.check_backward_options)
+
+    @condition.retry(3)
+    def test_backward_cpu(self):
+        self.check_backward(self.x, self.gy)
+
+    @attr.gpu
+    @condition.retry(3)
+    def test_backward_gpu(self):
+        self.check_backward(cuda.to_gpu(self.x), cuda.to_gpu(self.gy))
+
+    def check_backward_consistency_regression(self, x_data, gy_data):
+        # Regression test to two-dimensional unpooling layer.
+
+        ndim = len(self.dims)
+        if ndim != 2:
+            return
+
+        ksize = self.ksize
+        stride = self.stride
+        pad = self.pad
+        xp = cuda.get_array_module(x_data)
+
+        # Backward computation for N-dimensional unpooling layer.
+        x_nd = chainer.Variable(xp.array(x_data))
+        func_nd = functions.UnpoolingND(ndim, ksize, stride=stride,
+                                        pad=pad, cover_all=self.cover_all)
+        y_nd = func_nd(x_nd)
+        y_nd.grad = gy_data
+        y_nd.backward()
+
+        # Backward computation for two-dimensional unpooling layer.
+        x_2d = chainer.Variable(xp.array(x_data))
+        func_2d = functions.Unpooling2D(ksize, stride=stride, pad=pad,
+                                        cover_all=self.cover_all)
+        y_2d = func_2d(x_2d)
+        y_2d.grad = gy_data
+        y_2d.backward()
+
+        # Test that the two result gradients are close enough.
+        opt = copy.copy(self.check_backward_options)
```

Comment:
Ah-, it is unnecessary. I added ``copy`` to remove ``dtype`` key in order not to modify the original ``check_backward_options``, but it is actually set up for each test case. Just mistake.